src/Microsoft.Extensions.Logging.Log4Net.AspNetCore/Log4NetProvider.cs

```diff
@@ -224,7 +228,7 @@ private static Assembly GetCallingAssemblyFromStartup()
 
 				if (string.Equals(type?.Name, "Startup", StringComparison.OrdinalIgnoreCase))
 				{
-					return type.Assembly;
+					return type?.Assembly;
```

Comment:
This condition will never be hit, as previous `string.Equals()` already is doing the verification. Please, remove it. :)